carbon-design-system/carbon-addons-iot-react
# fix(DashboardEditor): selection styling and custom card rendering

Closes #1738 

**Summary**

- Selection style and interactions were not adhering to the UX design
  - Fixed styling by adding a black border when selected, while still allowing the blue focus border
  - Fixed selection to happen onClick of the card and onKeypress of 'Enter'
- Custom cards were not rendering at all

**Change List (commits, features, bugs, etc)**

- Adds custom card default icon
- Render CUSTOM cards by omitting the `content` prop and rendering the `content` instead
- Updated the `renderCardPreview` function to also take in the `commonCardProps` defined by the editor, in case the user doesn't want to recreate some of them
  - By updating the props, I was able to simplify the card mapping and rendering
- Changed card selection to only occur on click of the card OR when the card is focused and the 'Enter' key is pressed
- Changed card selection styling to a black border to match the react-grid resizing styling
- Updated DashboardEditor stories to include 2 types of custom cards: 1 with a custom component function, and 1 with a custom card content strings or elements
- Added left margin to keep the first and farthest left card's title aligned with the PageTitleBar header title (as instructed by UX)



**Acceptance Test (how to verify the PR)**

- Go to any DashboardEditor story and try variations of clicking cards, and tabbing over to cards and pressing 'Enter' to select
- Go to the DashboardEditor - initialValue story and see that both custom cards are rendering correctly. Open storybook locally and try some variations of passing custom components through the `content` prop
